### ISSUE TITLE
Prevent Content Delivery Manager from reverting feature configuration

### ISSUE THREAD

**Issue by dyanikoglu** [riverar/mach2#11]
enable 10727725 -v 1
enable 13849566 -v 1

These two codes are used for enabling "Sets" support for Windows Skip Ahead Insider Preview updates. After enabling these two flags, restarting computer makes them enabled. After next restart, changes are reverting back and Sets support gets disabled.

**Comment by dgadelha** [comment]
Any updates on this, @riverar?

**Comment by riverar** [comment]
@dgadelha @dyanikoglu Hey, sorry for the delay. When I last looked at this, Windows 10's Content Delivery Manager was the culprit. (It receives programming from the mothership on what features should be turned on/off.) This requires some revisiting, perhaps there's an off switch.

**Comment by dyanikoglu** [comment]
Hi @riverar,

Thanks for the reply, I'm looking forward to further updates about this issue.

**Comment by dgadelha** [comment]
An alternative would be installing mach2 as a service (or just a simple scheduled task) to re-apply the user-defined properties (enabled/disabled, with their respective variants) at every startup. About Content Delivery Manager, what about this _.reg_ file?

```
Windows Registry Editor Version 5.00

[HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\ContentDeliveryManager]
"FeatureManagementEnabled"=dword:00000000
"ContentDeliveryAllowed"=dword:00000000
```

**Comment by riverar** [comment]
@dgadelha I believe I disassembled CDM a long time ago and noticed those reg values are ignored if you're a non-Microsoft insider/build is WIP Slow/Fast. I'll dig up the details again.

**Comment by dyanikoglu** [comment]
@dgadelha I can confirm this method also does not work. After a few shutdown/power on actions, the flags are reverting themselves back.

**Comment by Biswa96** [comment]
@riverar How did you find that CDM is "the culprit"? I've found something different. In build 17692, the struct variable changed:
1. In your mach2: ` WNF_STATE_NAME wnf_machine_store{ 0xA3BC7C75, 0x418A073A };`
2. In build 17692: ` WNF_STATE_NAME wnf_machine_store{ 0xA3BC9835, 0x418A073A };`

I found that in wsl.exe. It will be helpful if you provide more details about the dll/exe file you've done reverse engineering.

**Comment by Biswa96** [comment]
I have done something with CDM in my [WinLight repo](https://github.com/Biswa96/WinLight). One test can be done. Try to disable all the `SubscribedContent` DWORD values in `HKCU\SOFTWARE\Microsoft\Windows\CurrentVersion\ContentDeliveryManager` registry key.

**Comment by Kiraxelle** [comment]
I had the same problem, I don't think that the problem comes with Content Delivery Manager (I think I have neither the service nor the process in my task manager), but all the modifications I performed with Mach2 were reinitialized after several days.

I found a solution using the task scheduler : I create a scheduled task for running Powershell (with maximum rights), and I add as argument "Mach2 disable ********", where ******* is the code of the hidden feature I want to disable.

This task runs at each startup, and is repeated every minute.

- It works
- It is invisible.
- It does not take CPU performences
- There is no sideeffects or bad interactions with someting else.

I think it can help those are in same case.
